app/services/filter_cms.rb

```diff
@@ -0,0 +1,49 @@
+class FilterCms
+  @version = "" # to store version of a url
+  class << self
+
+    def start_filter_wordpress_sites(urls_data)
+      url_html_version_map = Hash.new{ |h,k| h[k] = [] }
+      multi_threads = []
+      urls_data.each do |url|
+        multi_threads <<  Thread.new() {
+          begin
+            html = Nokogiri::HTML.parse(RestClient.get url)
+            if check_wordpress_in_meta(html)
+              url_html_version_map[url] = [html, @version]
```

Comment:
why have you used the @variable.
you could've done this instead

```ruby
version = check_wordpress_in_meta(html) # this can either return the version or nil
url_html_version_map[url] = [html, version] if version
```

Reply:
also, try to not use arrays, instead use a map to store the html and version like
`url_html_version_map[url] = {:html => html, :version => version} if version`